### PR TITLE
SDIT-405: 🐛 Fix compare index as Prison API rejecting max int query

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: hmpps-prisoner-search-indexer/src/main/resources/application.yml [hunk 1]
```diff
@@ -89,4 +89,4 @@ hmpps.sqs:
 
 index-build:
   page-size: 1000
-  complete-threshold: 700000
+  complete-threshold: 750000
```
Comment (petergphillips) [inline hmpps-prisoner-search-indexer/src/main/resources/application.yml:92]: bumped this as 784,627 current in production

File: hmpps-prisoner-search-indexer/helm_deploy/values-dev.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ generic-service:
     API_BASE_URL_PRISON_API: "https://prison-api-dev.prison.service.justice.gov.uk"
     API_BASE_URL_RESTRICTED_PATIENTS: "https://restricted-patients-api-dev.hmpps.service.justice.gov.uk"
     API_BASE_URL_INCENTIVES: "https://incentives-api-dev.hmpps.service.justice.gov.uk"
+    INDEX_BUILD_COMPLETE_THRESHOLD: 500000
```
Comment (petergphillips) [inline hmpps-prisoner-search-indexer/helm_deploy/values-dev.yaml:14]: less than 600,000 in production so wasn't hitting threshold

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/CompareIndexService.kt [hunk 3]
```diff
@@ -69,7 +69,7 @@ class CompareIndexService(
     }
   }
   fun compareIndex(): Pair<List<String>, List<String>> {
-    val allNomis = nomisService.getPrisonerNumbers(0, Int.MAX_VALUE).sorted()
+    val allNomis = nomisService.getPrisonerNumbers(0, 10000000).sorted()
```
Comment (petergphillips) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/CompareIndexService.kt:72]: this was being blocked by the new Prison API cloud platform firewall.  No reason to have it so high anyway

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**